### PR TITLE
Added multiple measurement support to projectQ backend

### PR DESCRIPTION
**Pull request by QFer**
* Added additional multiple measurement support for projectQ
  * get_probabilities_multiple_measurement
  * unit tests for multiple_measurement
* Changed 2018 to 2022 in copyright notices
* Version updated to 2.0.0 because of major changes in result structures for the new api-version
* Added documentation for multiple measurement (api) (class docstring, get_result, get_raw_data_from_result)
* Changed description of measurement_mask in result structure in api.get_result docstring
* Added qi_results.py to readthedoc

### CODE REVIEW

File: src/quantuminspire/projectq/backend_qx.py [hunk 1]
```diff
@@ -346,6 +346,39 @@ def _physical_to_simulated(self, physical_qubit_id: int) -> int:
...
+    def _simulated_to_logical(self, simulated_qubit_id: int) -> int:
+        """
+        Return the logical qubit id given the allocated location on the simulated backend.
+
+        :param simulated_qubit_id: the allocated location on the simulated backend.
+
+        :return:
+            Logical bit position of simulated qubit.
+        """
+        # First step: simulated to physical
```
Comment (tomrijnbeek) [inline src/quantuminspire/projectq/backend_qx.py:358]: optional: splitting this method into two named methods means we don't need to use comments

File: src/quantuminspire/projectq/backend_qx.py [hunk 2]
```diff
@@ -645,19 +721,27 @@ def __init__(self, qubit_id: int) -> None:
 
         random_measurement = self._sample_measured_states_once()
 
-        for logical_qubit_id in self._measured_ids:
-            physical_qubit_id = self._logical_to_physical(logical_qubit_id)
-            sim_qubit_id = self._physical_to_simulated(physical_qubit_id)
+        if self._full_state_projection:
```
Comment (tomrijnbeek) [inline src/quantuminspire/projectq/backend_qx.py:724]: optional: this if-else block is identical to the block in [672-678](https://github.com/QuTech-Delft/quantuminspire/pull/140/files#diff-9c941475897669e0a7260eceed76b5f53188f6f66e8632068dd885934070c23aR672-R678), with the exception of the `measurement_block_index`. Is it worth extracting into a separate function?

**Review by tomrijnbeek — APPROVED** [review]

File: src/quantuminspire/projectq/backend_qx.py (hunk 1 above)
Comment (QFer) [reply src/quantuminspire/projectq/backend_qx.py:358]: I have split it

**Review by tomrijnbeek — APPROVED** [review]
Definitely an improvement to have smaller and easier to understand methods, thanks!

File: src/quantuminspire/projectq/backend_qx.py [hunk 3]
```diff
@@ -328,6 +328,25 @@ def _deallocate_qubit(self, index_to_remove: int) -> None:
...
 
+    def _logical_to_physical(self, logical_qubit_id: int) -> int:
+        """Return the physical location of the qubit with the given logical id.
+
+        :param logical_qubit_id: ID of the logical qubit whose position should be returned.
+
+        :return:
+            Physical position of logical qubit with id qb_id.
+        """
+        if self.main_engine.mapper is not None:
```
[the remark this answers is not in the document]
Comment (QFer) [reply src/quantuminspire/projectq/backend_qx.py:339]: I may do this as a boyscouting action on a different activity

**Pull request merged**